packages/core/src/components/essentials/legend.ts

```diff
@@ -268,6 +268,7 @@ export class Legend extends Component {
 				const clickedItemData = clickedItem.datum() as any;
 
 				self.model.toggleDataLabel(clickedItemData.name);
+				self.model.updateSelectedGroups();
```

Comment:
since `updateSelectedGroups` is called every time `toggleDataLabel` is called, would it make sense to call `updateSelectedGroups` inside `toggleDataLabel` instead?

Reply:
I tried that, but seems the `selectedGroups` only shows former selected legends inside `toggleDataLabel` (no current clicked legend).